src/test/ui/where-clauses/where-clauses-method-unsatisfied.stderr

```diff
@@ -2,7 +2,9 @@ error[E0277]: the trait bound `Bar: Eq` is not satisfied
   --> $DIR/where-clauses-method-unsatisfied.rs:18:14
    |
 LL |     x.equals(&x);
-   |              ^^ the trait `Eq` is not implemented for `Bar`
+   |       ------ ^^ the trait `Eq` is not implemented for `Bar`
+   |       |
+   |       required by a bound introduced by this call
```

Comment:
This case has a `MiscObligation`, so for some `where` clauses we need to do better, but at least we point at the relevant argument. The new span pointing at the call at least hints at the user that they should look at the `equals` implementation to figure out why the bound was added.